### ISSUE TITLE
[dar2js] Replace `Math.sqrt` and `Math.pow` with the `**` operator

### ISSUE THREAD

**Issue by ClementBeal** [dart-lang/sdk#55272]
Considering this program :

```dart
import 'dart:math';

void main() {
  print(sqrt(22));
  print(pow(22, 3));
}
```

The generated code is :

```js
      A.print(Math.sqrt(22));
      A.print(Math.pow(22, 3));
```

In the case of the `Math.sqrt` and `Math.pow`, we could use the `**` operator to save some space. The expected output would be :

```js
      A.print(22*.5);
      A.print(22**3);
```

**Comment by rakudrama** [comment]
The size impact will be very small since most Flutter apps contain only a few dozen calls to these functions.

This might still be a reasonable change provided
1. The replacement always produces the same value as the original. Is there something in the specification that states or implies that they are equivalent?
2. The replacement is not slower than the original on major browsers. My initial attempts at profiling show that `x**.5` is slower than `Math.sqrt(x)` on V8, but faster on JSC, so I am a bit ambivalent about making the change.

**Comment by ClementBeal** [comment]
Yes, it's very small. In a basic flutter app, I can count 27 `Math.pow()` and 23 `Math.sqrt()` so it's at most a gain of `400B`.

1. https://262.ecma-international.org/7.0/#sec-math.pow 
They say that `Math.pow` is returning the result of the `**` operator. The tiny difference is the `**` operator supports BitInt numbers.

2. I believe V8 is way more used than JSC. Probably not so interesting to make the change in this case
